src/AltinnCore/Common/Services/Implementation/InstanceAppSI.cs

```diff
@@ -118,6 +118,7 @@ public async Task<List<Instance>> GetInstances(string applicationId, string appl
         {
             List<Instance> instances = null;
             DataContractJsonSerializer serializer = new DataContractJsonSerializer(typeof(Instance));
+            applicationId = ApplicationHelper.GetFormattedApplicationId(applicationOwnerId, applicationId);
             string apiUrl = $"{_platformStorageSettings.ApiUrl}/instances?instanceOwnerId={instanceOwnerId}&applicationOwnerId={applicationOwnerId}&applicationId={applicationId}";
```

Comment:
The implementation of this query lacks functionality. It only has queries for instanceOwnerId or applicationOwnerId or applicationId. Not a combination of the three! So you  need to consider reworking the query method. ApplicationOwnerId is redundant if applicationId is present.

Reply:
Fixed